UOP1_Project/Assets/Scripts/Pool/Example/PoolableParticle.cs

```diff
@@ -0,0 +1,37 @@
+﻿using OP1.Pool;
+using System;
+using System.Collections;
+using UnityEngine;
+
+public class PoolableParticle : MonoBehaviour, IPoolable
+{
+	[SerializeField]
+	private ParticleSystem _particleSystem = default;
+
+	public void OnRequest()
+	{
+		gameObject.SetActive(true);
+	}
+
+	public void Play()
+	{
+		_particleSystem.Play();
+	}
+
+	public void OnReturn(Action onReturned)
+	{
+		StartCoroutine(DoReturn(onReturned));
+	}
+
+	IEnumerator DoReturn(Action onReturned)
+	{
+		if (_particleSystem.isPlaying)
+		{
+			yield return new WaitForSecondsRealtime(_particleSystem.main.duration - (_particleSystem.time % _particleSystem.main.duration));
```

Comment:
Same comment here, we should probably use `WaitForSeconds` rather than its -`Realtime` counterpart.

Reply:
Or maybe here it would be good to multiply it all by `Time.timeScale`, in addition to making the change above?